web/src/components/CommandPalette.tsx

```diff
@@ -0,0 +1,420 @@
+import React, { useEffect, useState } from "react";
+import { matchSorter } from "match-sorter";
+import {
+  Badge,
+  Box,
+  Button,
+  chakra,
+  Flex,
+  Highlight,
+  HStack,
+  IconButton,
+  Input,
+  InputGroup,
+  InputRightElement,
+  Kbd,
+  Modal,
+  ModalBody,
+  ModalContent,
+  ModalHeader,
+  Text,
+  ModalOverlay,
+  ModalProps,
+  Spinner,
+  Tag,
+  Tooltip,
+  IconProps,
+  ComponentWithAs,
+  useUpdateEffect,
+} from "@chakra-ui/react";
+import { useRef } from "react";
+import {
+  AddIcon,
+  ArrowForwardIcon,
+  ArrowRightIcon,
+  CheckIcon,
+  DeleteIcon,
+  EditIcon,
+  PlusSquareIcon,
+  StarIcon,
+} from "@chakra-ui/icons";
+import { useNavigate, useRouter } from "react-location";
+import { GitCompareOutline } from "./icons/Icons";
+import { useListUserAccessRules } from "../utils/backend-client/end-user/end-user";
+import { ProviderIcon } from "./icons/providerIcon";
+import {
+  userListFavorites,
+  useUserListFavorites,
+} from "../utils/backend-client/default/default";
+
+type ICommand = {
+  name: string;
+  icon: ComponentWithAs<"svg", IconProps>;
+  action: () => void;
+  isAdminOnly?: boolean;
+  type?: "favorite";
+};
+
+const EditActionModal = ({
```

Comment:
I suspect this was copied over from power editor, I think we need to rename this for the context here
```suggestion
const CommandPallet = ({
```